src/guides/resolve-zil-domain.md

```diff
@@ -0,0 +1,244 @@
+---
+description: Simplest step-by-step guide on how to resolve .zil domain
+---
+
+# .zil Integration
+
+In order to resolve a .zil domain, we will need to satisfy three steps below
+
+* Get a namehash of a domain
+* Get resolver contract address from a domain
+* Query resolver contract to fetch the records
+
+Let's visualize the resolution process using some of the simplest tools web developer has: knowledge of HTML and js.
+
+### Initialize the project folder
+
+As has been said above all we need is to create a folder and two files index.html and index.js respectively
+
+```text
+mkdir unstoppable-zil-resolution
+cd unstoppable-zil-resolution
+touch index.html index.js
+```
+
+Let's use some blank HTML page. We have connected our empty index.js as well as a CDN library of [**js-sha256**](https://www.npmjs.com/package/js-sha256?utm_source=cdnjs&utm_medium=cdnjs_link&utm_campaign=cdnjs_library) for future namehashing step.
+
+```markup
+<!DOCTYPE html>
+<html lang="en">
+    <head>
+        <meta charset="utf-8">
+        <title>Basic .zil integration</title>
+    </head>
+    <body>
+
+      <script 
+        src="https://cdnjs.cloudflare.com/ajax/libs/js-sha256/0.9.0/sha256.min.js" 
+        integrity="sha512-szJ5FSo9hEmXXe7b5AUVtn/WnL8a5VofnFeYC2i2z03uS2LhAch7ewNLbl5flsEmTTimMN0enBZg/3sQ+YOSzQ==" 
+        crossorigin="anonymous"></script>
+
+      <script src="index.js"></script>
+    </body>
+</html>
+```
+
+Don't forget to add an input field and a button that will trigger the resolution process
+
+```markup
+      <input id="input" />
+      <button onclick="resolve()">Resolve</button>
+```
+
+#### As for our index.js file
+
+We are going to put some basic code to capture the text from the input field and print it in our console
+
+```javascript
+// <!-------> Resolving domain <!------->
+async function resolve() {
+  const userInput = document.getElementById("input").value;
+  console.log({ domain: userInput });
+}
+```
+
+### Taking a namehash
+
+Namehashing is an algorithm that converts a domain name in a classical format \(like example.crypto\) to a token id that Zilliqa contract can understand. 
+
+{% hint style="warning" %}
+It is important to know the difference between Zilliqa namehashing and [ERC-721](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-137.md#namehash-algorithm) which is a part of [EIP-137](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-137.md).
+In ZIL we use **sha256 from SHA-2**, instead of **keccak256** which is used across Ethereum chain
+{% endhint %}
+
+To do so we need to split the domain by "." character to get each label and then reduce the label's array with a sha256 hashing of an accumulator and next label starting from the end. 
+
+For the purposes of keeping this tutorial short, instead of going into the details of this process, we are going to use the namehash function with some adaptation to the hashing library
```

Comment:
```suggestion
For the purposes of keeping this tutorial short, instead of going into the details of this process, we are going to use the namehash function with some adaptation to the hashing library.
```

Btw, what do you mean by the adaptation?

Reply:
shaWrapper function was not initially planned and I had to include it there due to cdn library limitation. the sha library on cdn doesn't really convert hexString to an array of hex values which is required for namehashing